include/nitro/broken_options/option/group.hpp

```diff
@@ -41,34 +44,190 @@ namespace broken_options
     class group
     {
     public:
-        group(const std::string& name, const std::string& description = std::string(""))
-        : name_(name), description_(description)
+        std::string name;
+        std::string description;
+
+    public:
+        group(std::set<std::string>& all_argument_names, const int& position,
+              const std::string& name, const std::string& description = std::string(""))
+        : name(name), description(description), all_argument_names_(all_argument_names),
+          position_(position)
         {
         }
 
-        void add(base& option)
+        group& subgroup(const std::string& name, const std::string& description = std::string(""))
         {
-            options_.emplace(option.name(), std::ref(option));
+            if (std::find(sub_groups_.begin(), sub_groups_.end(), name) == sub_groups_.end())
+                sub_groups_.emplace_back(
+                    group(all_argument_names_, position_ + 1, name, description));
+
+            return *std::find(sub_groups_.begin(), sub_groups_.end(), name);
         }
 
-        void usage(std::ostream& s)
+        void usage(std::ostream& s) const
         {
-            s << name_ << ":" << std::endl;
-            if (description_.size())
+            s << std::endl;
+            for (int i = 0; i < position_; ++i)
+                s << "  ";
+
+            s << name << ":" << std::endl;
+
+            if (!description.empty())
             {
-                s << std::endl << description_ << std::endl << std::endl;
+                s << std::endl << description << std::endl << std::endl;
             }
 
+            std::map<std::string, const nitro::broken_options::base&> print_options;
             for (auto& iter : options_)
+                print_options.emplace(iter.first, iter.second);
+            for (auto& iter : multi_options_)
```

Comment:
Use the `insert()` member with two iteratoris instead of the three loops

Reply:
Impossible due cast